src/main/scala/pirate/Usage.scala

```diff
@@ -35,6 +35,20 @@ object Usage {
     case ArgumentParser(meta, p) =>
       ArgumentInfo(meta.metavar, meta.description, info.multi)
   }
+
+  def explain[A](command: Command[A], fails: NonEmptyList[Parser[Unit]]): String =
+    ({
+      (fails.toList.collect { case FlagParser(meta, _) => meta.names.get.render } match {
+        case Nil => None
+        case ms  => ms.mkString("Missing flags: ", ", ", "").some
+      }) :: (fails.toList.collect { case ArgumentParser(meta, _) => meta.metavar.getOrElse("ARG") } match {
+        case Nil => None
+        case ms  => ms.mkString("Missing arguments: `", "`, `", "`").some
+      }) :: (fails.toList.collect { case CommandParser(sub) => sub.name } match {
+        case Nil => None
+        case ms  => ms.mkString("Missing commands: `", "`, `", "`").some
+      }) :: Nil
+    }.flatten.mkString("\n")) ++ printWith(command, DefaultUsageMode)
 }
```

Comment:
This section is self standing and can become as complex as it likes, but everything required for a good usage string is here.